### PR TITLE
Add support for octree labels

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR will supersede #2392 by adding support for octree labels. In order to get the labels working we still have our pesky `raw_to_displayed` function (renamed `image_converter` inside the octree). Whereas in #2392 this wasn't getting called at all, now this function gets called, but it is getting called inside the main thread after a chunk arrives from the worker thread. This leads to really bad performance as the function is slow!

One option would be to move it to the worker thread, another option would be to call the function on the entire array using dask map_blocks, and then compute will get triggered in the worker thread for us, or we wait until all of this can happen on the GPU and we can drop the `raw_to_displayed` entirely.

Curious @jni @DragaDoncila what you recommend as a next step.

I'll also note in testing this PR I had some problems with multiple multiscale layers which I'll look into elsewhere.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
Just to note that this is the example I'm working with right now - 

```python
file_name = 'data/camelyon16/tumor_001.zarr'
pyramid = [da.from_zarr(file_name + '/' + str(i)) for i in range(10)]

def theshold_block(block):
    return block[:, :, 0] > 220
     
segmented = [ar.map_blocks(theshold_block, drop_axis=2, chunks=ar.chunksize[:-1], dtype=np.int)[:ar.shape[0], :ar.shape[1]] for ar in pyramid]


# create an empty viewer
viewer = napari.Viewer()

# add the pyramid
# viewer.add_image(pyramid, name='slide', multiscale=True)
viewer.add_labels(segmented, name='thresholded')
napari.run()
```

If I turn off the `image_converter` it's actually still slow, so I should make myself a labels zarr that i can load in to compare.

I'm wondering if there might be more we can do, with calling `dask.compute` to speed things up too, but that is maybe separate ... i'll look into all this more!

**Comment by sofroniewn** [comment]
Ok, I think this is ready for review now @jni. It puts the raw to displayed conversion in the vispy visual which means that it happens in the main thread.

Right now painting doesn't really work because the tiles are cached and so don't update. I can work on that in a follow up PR, but it would be good to merge this one first!

File: napari/_vispy/experimental/_tests/test_vispy_tiled_image.py [hunk 1]
```diff
@@ -221,3 +221,48 @@ def test_tiled_single_scale(qtbot, monkeypatch, make_napari_viewer):
     np.testing.assert_allclose(
         screenshot[-screen_offset, -screen_offset], target_center
     )
+
+
+@pytest.mark.async_only
+@pytest.mark.skip("NAPARI_OCTREE env var cannot be dynamically set")
```
Comment (jni) [inline napari/_vispy/experimental/_tests/test_vispy_tiled_image.py:227]: @sofroniewn what does this mean? ie what's the point of everything else :joy:

Comment (jni) [reply napari/_vispy/experimental/_tests/test_vispy_tiled_image.py:227]: ie: why not skip *only* if NAPARI_OCTREE is not externally set?

**Review by jni — APPROVED** [review]

File: napari/_vispy/experimental/_tests/test_vispy_tiled_image.py (hunk 1 above)
Comment (sofroniewn) [reply napari/_vispy/experimental/_tests/test_vispy_tiled_image.py:227]: It's really just a place holder for now until we figure out how to run these on CI, but I should make a PR that allows them to be run locally if the NAPARI_OCTREE is externally set (same as what you said). I'll do that in a followup for all tests in this file

**Pull request merged**